docs/general/Mainnet/node_level_architecture.md

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 2
+---
+
+# Node-level architecture
+
+## Node Types
+
+### Full node
+
+A full node provides the entire functionality of a Movement node:
+
+- ingresses transactions into a local mempool
+- forms proto-blocks from the transactions in the mempool and forwards them
+  to the [DA light node](#da-light-node)
+- runs executions of blocks sequenced by the DA,
+  receiving them through the DA light node.
+- provides views on the optimistic and settled (finalized) states via
+  Aptos RPC API endpoints.
+- (in a future decentralized network) can operate as a [validator node](#validator-node) with a key from the trusted validator set.
+
```

Comment:
Do we want to maybe look at what it means to run each one of these nodes practically here also?

Reply:
this seems a requirements question which should be answered by @rolandoesparza